aptos-move/block-executor/src/executor.rs

```diff
@@ -272,7 +293,41 @@ where
                 }
             }
 
-            if let Some(err) = last_input_output.execution_error(txn_idx) {
+            let group_metadata_ops = last_input_output.group_metadata_ops(txn_idx);
```

Comment:
why is this in prepare_and_queue_commit_ready_txns, and not in materialize_txn_commit ?

Reply:
we want to call commit_group single threaded, but it's not doing any serialization or value copies (still under Arc). The tags might be copied but we can intern / optimize that later if it becomes an issue on the sequential commit path. Will add a comment as well.

Reply:
I agree, wrote a long comment with a dissertation about design principle here (prepare_and_queue does light logic with flat combining, i.e. no concurrency to avoid overhead and be simple, while materialization can then be embarassingly parallelizable and do heavier, e.g. value patching, etc, logic)